### PR TITLE
August dependencies

### LINKED ISSUE #5438: August Dependency Updates
We should try to review and merge in dependency updates on a monthly basis. Until this is automated more, here is a guide:

- Issue summary should contain a list of the dependabot pulls in scope for reference, however, the actual work should be done in a single PR to minimize composer conflicts.
- Dependencies related to our frontend build tools (not executed on the websites) are usually safe to merge as-is but might cause issue with other packages.
- As a separate PR is created with the dependency update, the related dependabot version should be closed referencing the monthly PR.
- Periodically check [locked dependencies](https://github.com/uiowa/uiowa/blob/main/README.md#locked-packages) for new versions. Make notes about why a dependency was/is locked in the repo Readme.
- Review the changelog for each version we are jumping to and the versions in-between. Make note of certain functionality in our codebase that is using the functionality for testing instructions later.
- Use https://github.com/uiowa/uiowa#updating-dependencies as a guide to test for update hooks, configuration changes and commit. This command is helpful if there are a lot of configuration changes across feature/site splits: https://github.com/uiowa/uiowa/blob/main/blt/src/Blt/Plugin/Commands/ConfigSplitCommands.php
- If a particular dependency is too big to go out along with others, split it off from the PR as a separate issue and introduce the "why" in parking lot.
- Update hooks and requires cache rebuild type updates may require off-hours deployment.
- Once this issue is completed, copy these notes to the next month's issue and introduce it in parking lot.

## Updates in-scope
- [x] https://github.com/uiowa/uiowa/pull/5533
- [x] https://github.com/uiowa/uiowa/pull/5532
- [x] https://github.com/uiowa/uiowa/pull/5531
- [x] https://github.com/uiowa/uiowa/pull/5530
- [x] https://github.com/uiowa/uiowa/pull/5517
- [x] https://github.com/uiowa/uiowa/pull/5535
- [x] https://github.com/uiowa/uiowa/pull/5515
- [x] https://github.com/uiowa/uiowa/pull/5514
- [x] https://github.com/uiowa/uiowa/pull/5513
- [x] https://github.com/uiowa/uiowa/pull/5539
- ~~https://github.com/uiowa/uiowa/pull/5511~~
- [x] https://github.com/uiowa/uiowa/pull/5510
- ~~https://github.com/uiowa/uiowa/pull/5509~~
- ~~https://github.com/uiowa/uiowa/pull/5508~~
- [x] https://github.com/uiowa/uiowa/pull/5419
- [x] https://github.com/uiowa/uiowa/pull/5323
- [x] https://github.com/uiowa/uiowa/pull/5215

## Possible locked dependency updates

- cshs
- menu_link_weight
- reroute_email

## Separate issues

- https://github.com/uiowa/uiowa/pull/5324 will be addressed in #5467
- https://github.com/uiowa/uiowa/pull/5205 will be addressed in #5332

### PR DESCRIPTION
**Pull request by joewhitsitt**
Resolves https://github.com/uiowa/uiowa/issues/5438

# How to test

- [x] basically `dsa` a bunch of sites to check the functionality and config import of the steps below. checkout this sweet up to date list: https://iowaweb.slack.com/archives/C0164FNN1PV/p1659627793779479
- [ ] if you want to go further and check that I got all of the configuration exported correctly, consider running these commands https://github.com/uiowa/uiowa/blob/main/blt/src/Blt/Plugin/Commands/ConfigSplitCommands.php
  - keep in mind that CS2 isn't perfect and you will get some crap exports. if they don't relate to these modules, you can probably disregard them. if they are close, consider doing more checks surrounding it and manually massage the configuration to perfection. 
    - **How, where, and why do we run these?**
      -  `blt uiowa:usplits:feature`, `blt uiowa:usplits:migrate`, `blt uiowa:usplits:site`

- [x] inline_entity_form, test article author functionality as well as sliders. We are using a patch that still applies - "Custom blocks created in inline forms should not be reusable" which was introduced in https://github.com/uiowa/uiowa/pull/1991 **no known config changes**
- [x] pathauto, test new page creation nested in the menu system, updating an existing node alias, and going to /admin/config/sitenow/uiowa-articles and changing the path and then inspecting article content to see that their paths have been updated. **update hook, config changes** - `Update node type conditions from node_type to entity_bundle.`
- [x] Test article config form changes on sppa.uiowa.edu, hr.uiowa.edu, iisc.uiowa.edu to make sure there aren't config import issues.  
- [x] google_tag, there were **no known config changes** however, "Changes to default container settings and insertion conditions only apply to new containers," and `google_tag.container.*` is ignored. Might be worth re-saving site containers after deployment.
- [x] paragraphs, test a v2 website, v3 collections, person contact, etc. to make sure functionality still works. **no known config changes** however, removes patch related to https://github.com/uiowa/uiowa/pull/2112
- [x] moderated_content_bulk_publish, test to make sure bulk operations on the content overview work as before (changing to publish, changes the moderation state). this module now requires views_bulk_operations which we didn't have before. This addition in turn, adds new vbo functionality to fragments, which I have turned off for now. **update hook, configuration changes**
- [x] config_inspector - d10 compatibility, changing from jquery.once to drupal.once. don't worry, we still have schema issues with our config but they are unrelated to this update. **no known config changes**
- [x] menu_per_role - **config changes** make sure the test instructions from https://github.com/uiowa/uiowa/pull/3271 still work. See if we can move some code into settings https://github.com/uiowa/uiowa/issues/5537
- [x] file_mdm - **no known config changes**. doesn't seem to fix https://github.com/uiowa/uiowa/pull/5390#issuecomment-1188356640 so that's a thing. Not sure how to test this otherwise. We do have some custom code that relies on this module https://github.com/uiowa/uiowa/blob/main/docroot/modules/custom/sitenow_media_wysiwyg/sitenow_media_wysiwyg.module#L104
- [x] layout_builder_restrictions **update hook, config changes** moves away from white/black-listing labels. https://www.drupal.org/node/3179983 - should probably do a large `dsa` on sites to check for import errors.
- [x] cshs - **no known config changes** make sure https://github.com/uiowa/uiowa/issues/4747 isn't happening again and that other functionality still is working.
- [x] menu_link_weight - **no known config changes** make sure https://github.com/uiowa/uiowa/pull/4523#issuecomment-1004309342 isn't happening and I'd say you are good.
- [x] reroute_email - **config changes** I think unless there are import errors (in any environment) this can just go out. it was locked to fix for [specific reasons](https://github.com/uiowa/uiowa/pull/4741#issuecomment-1040606450) and this isn't used in production 

# Next steps

- This should be part of a scheduled deployment.

### CODE REVIEW

**Comment by joewhitsitt** [comment]
ugh. messed up merge

**Comment by joewhitsitt** [comment]
layout_builder_restrictions is the one I would probably test the most. others on the team are more familiar with what blocks we are allowing where on what sites and if the updated config works out.

**Comment by bspeare** [comment]
Noticing that for `moderated_content_bulk_publish` the “Are you sure you want to publish this?” message is back: https://www.drupal.org/project/moderated_content_bulk_publish/issues/3185848.  Looks like the setting needs to be changed on `/admin/config/content/moderated-content-bulk-publish`

**Comment by bspeare** [comment]
I'm not seeing this "Units" field on the prod hr.uiowa.edu site.  Does anyone know how/where this should be appearing on the article node creation/edit form?

Screen Shot 2022-08-10 at 2 47 16 PM

**Comment by bspeare** [comment]
@e-marie-w and I worked on this today.  We made it through most of it but have added a few questions in bold under the unchecked items above (https://github.com/uiowa/uiowa/pull/5527#issue-1326424567).

**Comment by briand44** [comment]
>What's the best way to test that function that rotates for exif metadata?

wouldn't this be to upload a new image (jpeg) through the WYSIWYG that has an orientation of something other than 1 or 2 and verify that the image after uploading is the correct orientation and has a value of 1? You could comment the code out locally as well and test that nothing happens and the orientation doesn't change?

**Comment by briand44** [comment]
>What does re-saving site containers mean?

I assume it means saving this form on sites using GTM

https://admissions.uiowa.edu/admin/config/system/google-tag/manage/gtm?destination=/admin/config/system/google-tag

**Comment by bspeare** [comment]
> What's the best way to test that function that rotates for exif metadata?

I've tried this with a number of photos with EXIF data other than 1 or 2 and I see no change on local or prod.   Perhaps I don't have the perfect testing image.

**Comment by bspeare** [comment]
Tested `file_mdm` running through the function in xdebug with @e-marie-w and @briand44 and it changed exif 6 to exif 1.

**Comment by bspeare** [comment]
We resaved `/admin/config/system/google-tag` and got the message `Created 3 snippet files for Tippie Prod V3 container based on configuration.`  Is that the standard Google Tag Manager message?

**Comment by bspeare** [comment]
I did test "Test article config form changes" and found no issues and checked the box.  I found where to look for https://github.com/uiowa/uiowa/pull/5527#issuecomment-1211193495.

**Comment by bspeare** [comment]
Double checked https://github.com/uiowa/uiowa/pull/5527#issuecomment-1212136258 and that looks like the standard message when saving.

**Comment by bspeare** [comment]
I think this is probably ready to go.  The only thing we didn't run was the "if you want to go further and check" steps.

**Comment by joewhitsitt** [comment]
Coming back, it was a bit hard to follow the PR summary activity but I see where most comments align. 

>How, where, and why do we run these?
blt uiowa:usplits:feature, blt uiowa:usplits:migrate, blt uiowa:usplits:site

These would be run across all sites/feature/migrations locally to check for configuration changes related to the dependency updates. This is the large scale equivalent of "To ensure configuration is exported correctly, manually sync a site from production using Drush. Then run database updates and export any configuration changes. Add and commit the config changes and then run another blt dsa to check for any further config discrepancies." from https://github.com/uiowa/uiowa#updating-dependencies

Given how some of these splits are turned on the config results might not be accurate.

The resulting "good" config will prevent config import errors on deployment and allow splits to accept new changes to base functionality.

Besides this, were there any outstanding questions/concerns?

I plan on pulling this down one more time and coordinating a scheduled release with the team.

**Comment by bspeare** [comment]
Thanks for explaining that further @joewhitsitt.   We did the audit on `menu_per_role` and did not find any new functionality: https://github.com/uiowa/uiowa/issues/5537#issuecomment-1211235338.  Did we completely miss some changes/improvements that you were expecting?

**Comment by joewhitsitt** [comment]
Thanks for highlighting that. Missed it in my catch-up. Responded on that issue https://github.com/uiowa/uiowa/issues/5537#issuecomment-1215846890

**Review by bspeare — APPROVED** [review]
Nice work!
